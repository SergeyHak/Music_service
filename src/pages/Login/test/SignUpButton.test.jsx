import { SignupButton } from "./SignUoButton";
import { screen, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

const dataTestId = "test-id";

describe("<SignupButton/>", () => {
  describe("Snapshots tests", () => {
    it("should render successfully and match snapshot", () => {
      const { container } = render(
        <SignupButton>Зарегистрироваться</SignupButton>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe("Attributes tests", () => {
    it('should set type="button" by default', () => {
      render(<SignupButton />);
      expect(screen.getByRole("button")).toHaveAttribute("type", "button");
    });
  });
  it("should set type attribute", () => {
    const type = "submit";
    render(<SignupButton type={type} />);

    expect(screen.getByRole("button")).toHaveAttribute("type", type);
  });

  describe("Callbacks tests", () => {
    it("should call 'onClick' prop", () => {
      const cb = jest.fn();

      render(<SignupButton onClick={cb} dataTestId={dataTestId} />);

      fireEvent.click(screen.getByTestId(dataTestId));

      expect(cb).toBeCalledWith({ name: "Name" });
    });
  });
});
