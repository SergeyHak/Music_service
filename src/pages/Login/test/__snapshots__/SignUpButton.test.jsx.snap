// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupButton/> Snapshots tests should render successfully and match snapshot 1`] = `
<div>
  <button
    class="sc-eDvSVe lnOQyS"
    type="button"
  >
    Зарегистрироваться
  </button>
</div>
`;
